tests/e2e/power_monitor_internal_test.go

```diff
@@ -124,40 +125,125 @@ func TestPowerMonitorInternal_RBAC_Reconciliation(t *testing.T) {
 			),
 		)
 	}
-	// then the following resources will be created
-	f.AssertResourceExists(testNs, "", &corev1.Namespace{})
 
-	// generate missing certs required in openshift
-	clusterIssuerName := "selfsigned-cluster-issuer"
-	caCertName := "power-monitor-ca"
-	caCertSecretName := "power-monitor-ca-secret"
-	pmIssuerName := "power-monitor-ca-issuer"
-	tlsCertName := powermonitor.SecretTLSCertName
 	tlsCertSecretName := powermonitor.SecretTLSCertName
-	f.DeployOpenshiftCerts(name, testNs, clusterIssuerName, caCertName, caCertSecretName, pmIssuerName, tlsCertName, tlsCertSecretName)
+	var caCertSource string
+
+	if Cluster == k8s.Kubernetes {
+		// For Kubernetes clusters, deploy cert-manager and dependencies
+		clusterIssuerName := "selfsigned-cluster-issuer"
+		caCertName := "power-monitor-ca"
+		caCertSecretName := "power-monitor-ca-secret"
+		pmIssuerName := "power-monitor-ca-issuer"
+		f.DeployOpenshiftCerts(
+			pmi.Name,
+			testNs,
+			clusterIssuerName,
+			caCertName,
+			caCertSecretName,
+			pmIssuerName,
+			tlsCertSecretName,
+			tlsCertSecretName,
+		)
+		caCertSource = caCertSecretName
+	} else {
+		f.WaitUntilPowerMonitorInternalCondition(pmi.Name, v1alpha1.Reconciled, v1alpha1.ConditionTrue)
+		f.WaitForOpenshiftCerts(pmi.Name, testNs, tlsCertSecretName)
```

Comment:
So it looks like you are waiting for this, which is good. I would rather use Cluster == k8s.OpenShift rather than use else. Because the certs being added are only automatically added by openshift as far as we know. So in other clusters like kubernetes we should expect to manually add these "openshift" certs.